### ISSUE TITLE
Remove unnecessary unsafe code warnings in 1.8.0

### ISSUE THREAD

**Issue by jerryxwu** [timescale/timescaledb-toolkit#462]
1.8.0 build scripts generate the following compiling warnings. They should be removed.

```
warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline/aggregation.rs:166:64
    |
165 | pub unsafe fn pipeline_stats_agg_support(input: Internal) -> Internal {
    | --------------------------------------------------------------------- because it's nested under this `unsafe` fn
166 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: `#[warn(unused_unsafe)]` on by default
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`
    = note: `#[allow(unsafe_op_in_unsafe_fn)]` on by default
warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline/aggregation.rs:260:64
    |
259 | pub unsafe fn pipeline_sum_support(input: Internal) -> Internal {
    | --------------------------------------------------------------- because it's nested under this `unsafe` fn
260 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`
warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline/aggregation.rs:350:64
    |
349 | pub unsafe fn pipeline_average_support(input: Internal) -> Internal {
    | ------------------------------------------------------------------- because it's nested under this `unsafe` fn
350 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`
warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline/aggregation.rs:437:64
    |
436 | pub unsafe fn pipeline_num_vals_support(input: Internal) -> Internal {
    | -------------------------------------------------------------------- because it's nested under this `unsafe` fn
437 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`
warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline/aggregation.rs:520:64
    |
519 | pub unsafe fn pipeline_counter_agg_support(input: Internal) -> Internal {
    | ----------------------------------------------------------------------- because it's nested under this `unsafe` fn
520 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`
warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline/aggregation.rs:603:64
    |
602 | pub unsafe fn pipeline_hyperloglog_support(input: Internal) -> Internal {
    | ----------------------------------------------------------------------- because it's nested under this `unsafe` fn
603 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`
warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline/aggregation.rs:676:64
    |
675 | pub unsafe fn pipeline_percentile_agg_support(input: Internal) -> Internal {
    | -------------------------------------------------------------------------- because it's nested under this `unsafe` fn
676 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`
warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline/expansion.rs:148:64
    |
147 | pub unsafe fn pipeline_materialize_support(input: pgx::Internal) -> pgx::Internal {
    | --------------------------------------------------------------------------------- because it's nested under this `unsafe` fn
148 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`
warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline.rs:171:64
    |
170 | pub unsafe fn pipeline_support(input: Internal) -> Internal {
    | ----------------------------------------------------------- because it's nested under this `unsafe` fn
171 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`
```

**Comment by rtwalker** [comment]
Just noting that these warnings appeared starting with Rust 1.61
```
~/t/t/extension (main)> rustup run 1.60 cargo b --features=pg14
    Blocking waiting for file lock on build directory
   Compiling timescaledb_toolkit v1.9.0-dev (/Users/rwalker/timescale/timescaledb-toolkit/extension)
    Finished dev [unoptimized + debuginfo] target(s) in 36.63s
```
```
~/t/t/extension (main)> rustup run 1.61 cargo b --features=pg14
    Blocking waiting for file lock on build directory
   Compiling timescaledb_toolkit v1.9.0-dev (/Users/rwalker/timescale/timescaledb-toolkit/extension)
warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline/aggregation.rs:166:64
    |
165 | pub unsafe fn pipeline_stats_agg_support(input: Internal) -> Internal {
    | --------------------------------------------------------------------- because it's nested under this `unsafe` fn
166 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: `#[warn(unused_unsafe)]` on by default
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`
    = note: `#[allow(unsafe_op_in_unsafe_fn)]` on by default

warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline/aggregation.rs:260:64
    |
259 | pub unsafe fn pipeline_sum_support(input: Internal) -> Internal {
    | --------------------------------------------------------------- because it's nested under this `unsafe` fn
260 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`

warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline/aggregation.rs:350:64
    |
349 | pub unsafe fn pipeline_average_support(input: Internal) -> Internal {
    | ------------------------------------------------------------------- because it's nested under this `unsafe` fn
350 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`

warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline/aggregation.rs:437:64
    |
436 | pub unsafe fn pipeline_num_vals_support(input: Internal) -> Internal {
    | -------------------------------------------------------------------- because it's nested under this `unsafe` fn
437 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`

warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline/aggregation.rs:520:64
    |
519 | pub unsafe fn pipeline_counter_agg_support(input: Internal) -> Internal {
    | ----------------------------------------------------------------------- because it's nested under this `unsafe` fn
520 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`

warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline/aggregation.rs:603:64
    |
602 | pub unsafe fn pipeline_hyperloglog_support(input: Internal) -> Internal {
    | ----------------------------------------------------------------------- because it's nested under this `unsafe` fn
603 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`

warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline/aggregation.rs:676:64
    |
675 | pub unsafe fn pipeline_percentile_agg_support(input: Internal) -> Internal {
    | -------------------------------------------------------------------------- because it's nested under this `unsafe` fn
676 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`

warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline/expansion.rs:148:64
    |
147 | pub unsafe fn pipeline_materialize_support(input: pgx::Internal) -> pgx::Internal {
    | --------------------------------------------------------------------------------- because it's nested under this `unsafe` fn
148 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`

warning: unnecessary `unsafe` block
   --> extension/src/time_vector/pipeline.rs:171:64
    |
170 | pub unsafe fn pipeline_support(input: Internal) -> Internal {
    | ----------------------------------------------------------- because it's nested under this `unsafe` fn
171 |     pipeline_support_helper(input, |old_pipeline, new_element| unsafe {
    |                                                                ^^^^^^ unnecessary `unsafe` block
    |
    = note: this `unsafe` block does contain unsafe operations, but those are already allowed in an `unsafe fn`

warning: `timescaledb_toolkit` (lib) generated 9 warnings
    Finished dev [unoptimized + debuginfo] target(s) in 17.40s
```

**Comment by rtwalker** [comment]
Closed by #473
